Description: avoid use of eslint during build
Author: Jonas Smedegaard <dr@jones.dk>
Last-Update: 2020-12-07
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/scripts/generate-browser-globals.js
+++ b/scripts/generate-browser-globals.js
@@ -6,7 +6,6 @@
 
 const fs = require("fs")
 const path = require("path")
-const { CLIEngine } = require("eslint")
 const { browser: originalGlobals } = require("globals")
 
 const targetFile = path.resolve(__dirname, "../lib/configs/_browser-globals.js")
@@ -33,7 +32,6 @@
     }
 }
 
-const linter = new CLIEngine({ fix: true })
 const rawCode = `/**
 * DON'T EDIT THIS FILE WHICH WAS GENERATED BY './scripts/generate-browser-globals.js'.
 */
@@ -41,8 +39,6 @@
 
 module.exports = ${JSON.stringify(globals, null, 4)}
 `
-const code =
-    linter.executeOnText(rawCode, "_browser-globals.js").results[0].output ||
-    rawCode
+const code = rawCode
 
 fs.writeFileSync(targetFile, code)
--- a/scripts/generate-configs.js
+++ b/scripts/generate-configs.js
@@ -6,7 +6,6 @@
 
 const fs = require("fs")
 const path = require("path")
-const { CLIEngine } = require("eslint")
 
 const targetFile = path.resolve(__dirname, "../lib/configs.js")
 
@@ -27,7 +26,3 @@
 }
 `
 )
-
-const linter = new CLIEngine({ fix: true })
-const result = linter.executeOnFiles([targetFile])
-CLIEngine.outputFixes(result)
--- a/scripts/generate-rules.js
+++ b/scripts/generate-rules.js
@@ -6,7 +6,6 @@
 
 const fs = require("fs")
 const path = require("path")
-const { CLIEngine } = require("eslint")
 
 const targetFile = path.resolve(__dirname, "../lib/rules.js")
 
@@ -33,7 +32,3 @@
 )
 `
 )
-
-const linter = new CLIEngine({ fix: true })
-const result = linter.executeOnFiles([targetFile])
-CLIEngine.outputFixes(result)
